### PR TITLE
fix: remove race condition between rabbitmq connection queue.consume

### PR DESCRIPTION
**Pull request by Akamaozu**
Should fix this issue: https://github.com/pagerinc/jackrabbit/issues/87

Changed `onMessage` from variable with anonymous function assigned to it to hoisted named function.

No need to change structure of the code, just `onMessage`'s type.

Also changed a nearby if/else to early exit style.

### CODE REVIEW

**Comment by Akamaozu** [comment]
Hi team @pagerinc

Test failing because of style rules.

image

Could rewrite this to match the style by moving `onMessage` above `queue.consume`'s actual logic, but wanted some clarity on if this is a rule you want enforced or should change / remove.

cc: @ryanhallcs @cilindrox

File: lib/queue.js [hunk 1]
```diff
@@ -30,16 +30,17 @@ const queue = (options) => {
...
-        }
 
-        const onMessage = (msg) => {
+        emitter.once('ready', () => {
+
+            const opts = Extend({}, DEFAULT_CONSUME_OPTIONS, consumeOptions);
+            channel.consume(emitter.name, onMessage, opts, onConsume);
+        });
+
+        function onMessage(msg){
```
Comment (cilindrox) [inline lib/queue.js]: You can appease the linter by [declaring a top-level `internals`][0] var, eg:

```
const internals = {};
```

and then you can attach this headless func to it, eg: 

```suggestion
        internals.onMessage: (msg) => {
```

And that should keep the linter happy :)

[0]: https://gist.github.com/hueniverse/a06f6315ea736ed1b46d#module-classes

**Comment by cilindrox** [comment]
Thanks @Akamaozu - We tend to prefer hapi's style for the most part, left a comment there.

cc @ryanhallcs

**Comment by Akamaozu** [comment]
Seems easier to move the logic after the anonymous function assignment to a variable

**Comment by cilindrox** [comment]
That too can work, pick your poison :)

**Comment by cilindrox** [comment]
thanks for this @Akamaozu !

**Comment by cilindrox** [comment]
Cut v4.8.1 with this fix.

**Pull request merged**